src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/InviteUsers/InviteOrganizationUsersCommand.cs

```diff
@@ -0,0 +1,250 @@
+
+        var invitesToSend = request.Invites
+            .SelectMany(invite => invite.Emails
+                .Where(email => !existingEmails.Contains(email))
+                .Select(email => OrganizationUserInviteDto.Create(email, invite, request.InviteOrganization.OrganizationId))
+            ).ToArray();
+
+        if (invitesToSend.Length == 0)
+        {
+            return new Failure<IEnumerable<OrganizationUser>>(NoUsersToInvite);
+        }
+
+        var validationResult = await inviteUsersValidation.ValidateAsync(new InviteUserOrganizationValidationRequest
+        {
+            Invites = invitesToSend.ToArray(),
+            InviteOrganization = request.InviteOrganization,
+            PerformedBy = request.PerformedBy,
+            PerformedAt = request.PerformedAt,
+            OccupiedPmSeats = await organizationUserRepository.GetOccupiedSeatCountByOrganizationIdAsync(request.InviteOrganization.OrganizationId),
+            OccupiedSmSeats = await organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(request.InviteOrganization.OrganizationId)
+        });
+
+        if (validationResult is Invalid<InviteUserOrganizationValidationRequest> invalid)
+        {
+            return new Failure<IEnumerable<OrganizationUser>>(invalid.ErrorMessageString);
+        }
+
+        var validatedRequest = validationResult as Valid<InviteUserOrganizationValidationRequest>;
+
+        var organizationUserCollection = invitesToSend
+            .Select(MapToDataModel(request.PerformedAt))
+            .ToArray();
+
+        var organization = await organizationRepository.GetByIdAsync(validatedRequest!.Value.InviteOrganization.OrganizationId);
+
+        try
+        {
+            await organizationUserRepository.CreateManyAsync(organizationUserCollection);
+
+            await AdjustPasswordManagerSeatsAsync(validatedRequest, organization);
+
+            await AdjustSecretsManagerSeatsAsync(validatedRequest, organization);
+
+            await SendAdditionalEmailsAsync(validatedRequest, organization);
+
+            await SendInvitesAsync(organizationUserCollection, organization);
+
+            await PublishReferenceEventAsync(validatedRequest, organization);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, FailedToInviteUsers);
+
+            await organizationUserRepository.DeleteManyAsync(organizationUserCollection.Select(x => x.OrganizationUser.Id));
+
+            await RevertSecretsManagerChangesAsync(validatedRequest, organization);
+
+            await RevertPasswordManagerChangesAsync(validatedRequest, organization);
+
+            return new Failure<IEnumerable<OrganizationUser>>(FailedToInviteUsers);
+        }
+
+        return new Success<IEnumerable<OrganizationUser>>(organizationUserCollection.Select(x => x.OrganizationUser));
+    }
+
+    private async Task RevertPasswordManagerChangesAsync(Valid<InviteUserOrganizationValidationRequest> valid, Organization organization)
```

Comment:
Keeping consistency on the name  helps readability. Other methods also
```suggestion
    private async Task RevertPasswordManagerChangesAsync(Valid<InviteUserOrganizationValidationRequest> validatedRequest, Organization organization)
```